### PR TITLE
docs(client): Troubleshooting

### PR DESCRIPTION
**Pull request by xni**
## Description
During the installation of dependencies for my Mac, I have encountered the problem with node-gyp. It looks to be know, and goes to the `hid` module. But at the moment it's easier to put it's solution to the readme file rather then submitting a PR to `hid`, going back and forth.

## Motivation and Context
Give a hand of help to the people to want to contribute.

## How Has This Been Tested?
Following that instruction helped me to have nOS up and running.

## Types of changes
Documentation

## Checklist:
- [* ] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** guidelines and confirm that my code follows the code style of this project.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ *] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: docs/nos-client/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Command: sh
+Arguments: -c prebuild-install || node-gyp rebuild
+Directory: /Users/nikitin/Development/nos-client/node_modules/node-hid
+Output:
+prebuild-install WARN install No prebuilt binaries found (target=10.6.0 runtime=node arch=x64 platform=darwin)
+```
+
+_Solution:_ `yarn global add node-gyp`.
+
+_Explanation:_ Some node.js modules does not have pre-compiled binaries for your version of OS/node.js. To build them you need to have node-gyp installed in the global scope.
```
Comment (mhuggins) [inline docs/nos-client/troubleshooting.md]: "does not" => "do not"

File: docs/nos-client/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+This page aims to provide some guidance in case then you have
+encountered an error, which is beyond our control.
+
+## `yarn install` fails
+
+**Case**: installing dependencies via `yarn install` fails to install prebuilt binaries,
+          resulting in the following error message:
+
+```
+error An unexpected error occurred: "/Users/nikitin/Development/nos-client/node_modules/node-hid: Command failed.
```
Comment (DalderupMaurice) [inline docs/nos-client/troubleshooting.md]: If you could replace your username with <username> or something ^^

**Comment by DalderupMaurice** [comment]
Looks good, besides that it's a new file so it has to be added to `sidebar.json`.
have a look right [here](https://github.com/nos/documentation#adding-new-documentation)
Or if you'd like I can smash it in afterward :)

File: docs/nos-client/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+Exit code: 127
+Command: sh
+Arguments: -c prebuild-install || node-gyp rebuild
+Directory: /Users/nikitin/Development/nos-client/node_modules/node-hid
```
Comment (DalderupMaurice) [inline docs/nos-client/troubleshooting.md]: Same here :P

File: docs/nos-client/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Troubleshooting
+
+Although nOS installation process is very easy and smooth in most
+of the cases, errors still may occur for some less popular configuratios.
```
Comment (DalderupMaurice) [inline docs/nos-client/troubleshooting.md]: configurations* typo :P

File: docs/nos-client/troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+This page aims to provide some guidance in case then you have
```
Comment (DalderupMaurice) [inline docs/nos-client/troubleshooting.md]: in case when?

**Comment by DalderupMaurice** [comment]
@xni what's the status? Would you like to resolve the comments? 
If you wish, we could finish it for you :)

**Comment by xni** [comment]
@DalderupMaurice , I apologize for the delay. All done! Thanks a lot for helping me with this PR!

File: website/sidebars.json [hunk 6]
```diff
@@ -5,7 +5,8 @@
       "nos-client/getting-started",
       "nos-client/api",
       "nos-client/implementation-example",
-      "nos-client/features"
+      "nos-client/features",
+      "nos-client/troubleshooting.md"
```
Comment (mhuggins) [inline website/sidebars.json]: @DalderupMaurice Will this work with the `.md` extension here?  I see the other lines don't include it, but maybe it's irrelevant.

Comment (xni) [reply website/sidebars.json]: Oh, no, you are right, completely my fault. Thanks for spotting that out! Fixed

File: website/sidebars.json [hunk 7]
```diff
@@ -5,7 +5,8 @@
       "nos-client/getting-started",
       "nos-client/api",
       "nos-client/implementation-example",
-      "nos-client/features"
+      "nos-client/features",
+      "nos-client/troubleshooting"
```
Comment (DalderupMaurice) [inline website/sidebars.json]: I'm sorry! It's actually going to need the id from the troubleshooting file, which is `installation-troubleshooting` in this case. But I'd suggest renaming the id to `troubleshooting`

I'll clarify this in the readme :)

Comment (xni) [reply website/sidebars.json]: Thank you! Done!

**Pull request merged**